src/Tests/dotnet.Tests/CommandTests/ToolRunCommandTests.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.DotNet.Cli;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.CommandFactory;
+using Microsoft.DotNet.InternalAbstractions;
+using Microsoft.DotNet.ToolManifest;
+using Microsoft.DotNet.ToolPackage;
+using Microsoft.DotNet.Tools.Tool.Run;
+using Microsoft.Extensions.DependencyModel.Tests;
+using Microsoft.Extensions.EnvironmentAbstractions;
+using NuGet.Frameworks;
+using NuGet.Versioning;
+
+namespace Microsoft.DotNet.Tests.Commands.Tool
+{
+    public class ToolRunCommandTests: SdkTest
+    {
+        private const string ManifestFilename = "dotnet-tools.json";
+        private DirectoryPath _nugetGlobalPackagesFolder;
+
+        public ToolRunCommandTests(ITestOutputHelper log) : base(log)
+        {
+            _nugetGlobalPackagesFolder = new DirectoryPath(NuGetGlobalPackagesFolder.GetLocation());
+        }
+
+        [Fact]
+        public void WhenRunWithRollForwardOptionItShouldIncludeRollForwardInNativeHost()
+        {
+            var parseResult = Parser.Instance.Parse($"dotnet tool run $TOOLCOMMAND$ --allow-roll-forward");
+
+            var toolRunCommand = new ToolRunCommand(parseResult);
+
+            (FilePath fakeExecutable, LocalToolsCommandResolver localToolsCommandResolver) = DefaultSetup("dotnet-a");
+            IEnumerable<string> testForwardArgument = Enumerable.Empty<string>();
+
+            var result = localToolsCommandResolver.Resolve(new CommandResolverArguments()
+            {
+                CommandName = "dotnet-a",
+                CommandArguments = (toolRunCommand._allowRollForward != false ? new List<string> { "--roll-forward", "Major" } : Enumerable.Empty<string>()).Concat(testForwardArgument)
```

Comment:
I'm not sure but it seems like you shouldn't need to conditionally be adding `--roll-forward Major` to the CommandArguments here.  I think the `LocalToolsCommandResolver` should be doing that based on what was parsed from the command.  It looks like the parseResult and the toolRunCommand are not actually used though, so I'm not sure what's going on here.

Reply:
Thank you so much. I updated these two tests in `ToolRunCommandTests` to use `ResolveStrict` method in the `LocalToolsCommandResolver`, which is the method to be used in `ToolRunCommand`. The state for roll forward is determined when constructing the `toolRunCommand`, get passed in the `GetPackageCommandSpecUsingMuxer` function in the `LocalToolsCommandResolver`, and the `--roll-forward Major` in the native host is added in the `MuxerCommandSpecMaker`. Sorry for being unclear in the test and hopefully the updated tests are clearer!